### PR TITLE
Corrections for the MVP

### PR DESCRIPTION
**Pull request by iinaki**
## Summary
With this PR I implement the corrections for the Conway's game. I'm going to be adding them below, in the changes section.

## Changes
- Replace `live_cells_do` which was a function that received a closure to do an operation using the position of the cells, with `live_cells_positions` which returns an iterator over the position of the live cells.
- Added a makefile that can run the tests, clippy, format and run the game. In the README you can find the instructions on how to use it.
- Added an action that checks if the tests pass, and also if clippy sends out a warning or if the code is not formated with cargo fmt. It runs on push and on the PRs of every branch.
- Added a rust-toolchain that specifies that the project uses the latest stable Rust version.
- Improved the code by replacing the double for in `neighbour_positions` for the `NEIGHBOUR_OFFSETS` constant. Also change an if to handle the `true` case first when both `true` and `false` cases need to be checked.

### CODE REVIEW

**Review by juanimedone — APPROVED** [review]
Looks better now!

File: src/game.rs [hunk 1]
```diff
@@ -66,10 +66,8 @@ impl Game {
     }
 
     /// Iterates over the live cells and calls the given closure with the position of each cell.
```
Comment (lazcanoluca) [inline src/game.rs]: I believe this comment is out of date now.

**Review by lazcanoluca — APPROVED** [review]
I noticed an out of date comment. Otherwise, send it 🚀 🚀

File: src/game.rs (hunk 1 above)
Comment (iinaki) [reply src/game.rs]: Truee, hadn't noticed that thanks!

Comment (iinaki) [reply src/game.rs]: Done!

**Pull request merged**